rust/lance/src/dataset/sql.rs

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use crate::datafusion::LanceTableProvider;
+use crate::Dataset;
+use arrow_array::RecordBatch;
+use datafusion::execution::SendableRecordBatchStream;
+use datafusion::prelude::SessionContext;
+use lance_core::Error;
+use snafu::location;
+use std::sync::Arc;
+
+/// Customize the params of dataset's sql API.
+#[derive(Clone, Debug)]
+pub struct SqlOptions {
+    /// the dataset to run the SQL query
+    pub(crate) dataset: Option<Dataset>,
+
+    /// the SQL query to run
+    pub(crate) sql: String,
```

Comment:
why is the SQL string a part of the SQL option? Maybe it makes more sense to call the struct `SqlBuilder` based on how it is used?